client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -77,26 +88,29 @@
      *                          This is used if the controller endpoints are static and can be directly accessed.
      *                      2. pravega://ip1:port1,ip2:port2,...
      *                          This is used to autodiscovery the controller endpoints from an initial controller list.
+     * @param connectionFactory The factory for connections to the pravega Server
      */
-    public ControllerImpl(final URI controllerURI) {
+    public ControllerImpl(final URI controllerURI, ConnectionFactory connectionFactory) {
```

Comment:
Yes, it doesn't belong here now. I actually originally implemented it on ReaderGroupImpl, but moved it here with an eye towards implementing:
https://github.com/pravega/pravega/issues/1496
and eventually:
https://github.com/pravega/pravega/issues/1436